src/to.frontend/to.frontend/Factories/IApplicationState.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+
+namespace to.frontend.Factories
+{
+    public interface IApplicationState
+    {
+        T Get<T>(string key);
+        void Set<T>(string key, T value);
+    }
+
+    public class ApplicationState : IApplicationState
+    {
+        private readonly Dictionary<string, object> _state;
+
+        public ApplicationState()
+        {
+            _state = new Dictionary<string, object>
```

Comment:
check if a concurrent dictionary should be used due to concurrency

Reply:
Concurrent dictionary is not required at the moment, could be added later on.